### PR TITLE
add query of project roles

### PR DESCRIPTION
**Pull request by sudan45**
Addresses:
- https://github.com/the-deep/server/issues/1485
## This PR doesn't introduce any:

- [ ] temporary files, auto-generated files or secret keys
- [ ] n+1 queries
- [ ] flake8 issues
- [ ] `print`
- [ ] typos
- [ ] unwanted comments

## This PR contains valid:

- [ ] tests
- [ ] permission checks (tests here too)
- [ ] translations

### CODE REVIEW

File: apps/project/schema.py [hunk 1]
```diff
@@ -650,3 +655,7 @@ def resolve_public_projects_by_region(*args, **kwargs):
     @staticmethod
     def resolve_user_pinned_project(root, info, **kwargs):
         return ProjectPinned.objects.filter(user=info.context.user)
+
+    @staticmethod
+    def resolve_project_role(root, info):
```
Comment (k9845) [inline apps/project/schema.py]:
```suggestion
    def resolve_project_role(root, info) -> QuerySet:
```

**Review by k9845 — COMMENTED** [review]
@sudan45 Let's also specify issue in pr.

File: apps/project/schema.py [hunk 2]
```diff
@@ -592,6 +596,7 @@ class Query:
     recent_projects = graphene.List(graphene.NonNull(ProjectDetailType))
     recent_activities = graphene.List(graphene.NonNull(RecentActivityType))
     project_explore_stats = graphene.Field(ProjectExploreStatType)
+    project_role = graphene.List(graphene.NonNull(ProjectRoleType))
```
Comment (thenav56) [inline apps/project/schema.py]:
```suggestion
    project_roles = graphene.List(graphene.NonNull(ProjectRoleType))
```

File: apps/project/schema.py [hunk 3]
```diff
@@ -202,6 +202,10 @@ class Meta:
 
     type = graphene.Field(ProjectRoleTypeEnum, required=True)
 
+    @classmethod
+    def queryset(cls):
+        return ProjectRole.objects.all()
+
```
Comment (thenav56) [inline apps/project/schema.py]: This is not required. We use queryset for permission. For normal field this is not required.

File: apps/project/schema.py [hunk 4]
```diff
@@ -650,3 +655,7 @@ def resolve_public_projects_by_region(*args, **kwargs):
     @staticmethod
     def resolve_user_pinned_project(root, info, **kwargs):
         return ProjectPinned.objects.filter(user=info.context.user)
+
+    @staticmethod
+    def resolve_project_role(root, info):
+        return ProjectRoleType.queryset()
```
Comment (thenav56) [inline apps/project/schema.py]: A basic ProjectRole.objects.all() should work here

**Review by thenav56 — CHANGES_REQUESTED** [review]

File: apps/project/schema.py [hunk 5]
```diff
@@ -592,6 +592,7 @@ class Query:
     recent_projects = graphene.List(graphene.NonNull(ProjectDetailType))
     recent_activities = graphene.List(graphene.NonNull(RecentActivityType))
     project_explore_stats = graphene.Field(ProjectExploreStatType)
+    project_roles = graphene.List(graphene.NonNull(ProjectRoleType))
```
Comment (thenav56) [inline apps/project/schema.py]:
```suggestion
    project_roles = graphene.List(graphene.NonNull(ProjectRoleType), required=True)
```

File: apps/project/tests/test_schemas.py [hunk 6]
```diff
@@ -1003,6 +1004,29 @@ def _query_check(project_id, **kwargs):
...
+        user = UserFactory.create()
+        # without login
+
+        self.query_check(query, assert_for_error=True)
+
+        # with login
+        self.force_login(user)
+        projectrole = ProjectRole.objects.count()
+        content = self.query_check(query)
+        self.assertEqual(len(content['data']['projectRoles']), projectrole)
```
Comment (thenav56) [inline apps/project/tests/test_schemas.py]:
```suggestion
        project_role_count = ProjectRole.objects.count()
        
        # without login
        self.query_check(query, assert_for_error=True)

        # with login
        self.force_login(user)
        content = self.query_check(query)
        self.assertEqual(len(content['data']['projectRoles']), project_role_count)
```

**Review by thenav56 — CHANGES_REQUESTED** [review]
Look good.

File: apps/project/tests/test_schemas.py (hunk 6 above)
Comment (susilnem) [reply apps/project/tests/test_schemas.py]: @thenav56 Can we make a factory for Roles and use that here?

Comment (thenav56) [reply apps/project/tests/test_schemas.py]: We have a static set of roles. For test, we have this function which generates here https://github.com/the-deep/server/blob/6a823454917cb15b7df7f4ee33552eaafe8bf027/utils/graphene/tests.py#L137-L189

**Review by thenav56 — DISMISSED** [review]

**Pull request merged**